components/layout_2020/flow/mod.rs

```diff
@@ -1409,27 +1409,24 @@ fn justify_self_alignment(containing_block: &ContainingBlock, free_space: Au) ->
 fn solve_inline_margins_for_in_flow_block_level(
     containing_block: &ContainingBlock,
     pbm: &PaddingBorderMargin,
-    inline_size: Length,
+    inline_size: Au,
 ) -> ((Au, Au), Au) {
-    let free_space =
-        containing_block.inline_size - pbm.padding_border_sums.inline - inline_size.into();
+    let free_space = containing_block.inline_size - pbm.padding_border_sums.inline - inline_size;
     let mut justification = Au::zero();
     let inline_margins = match (pbm.margin.inline_start, pbm.margin.inline_end) {
         (AuOrAuto::Auto, AuOrAuto::Auto) => {
             let start = Au::zero().max(free_space / 2);
             (start, free_space - start)
         },
-        (AuOrAuto::Auto, AuOrAuto::LengthPercentage(end)) => {
-            (Au::zero().max(free_space - end), end)
-        },
+        (AuOrAuto::Auto, AuOrAuto::LengthPercentage(end)) => (free_space - end, end),
         (AuOrAuto::LengthPercentage(start), AuOrAuto::Auto) => (start, free_space - start),
         (AuOrAuto::LengthPercentage(start), AuOrAuto::LengthPercentage(end)) => {
             // In the cases above, the free space is zero after taking 'auto' margins into account.
             // But here we may still have some free space to perform 'justify-self' alignment.
             // This aligns the margin box within the containing block, or in other words,
             // aligns the border box within the margin-shrunken containing block.
             let free_space = Au::zero().max(free_space - start - end);
-            justification = justify_self_alignment(containing_block, free_space);
+            let mut justification = justify_self_alignment(containing_block, free_space);
```

Comment:
Do not change this. This must be setting the existing variable, not create a local one.